### PR TITLE
VACMS-2323 Step By Step Component

### PR DESCRIPTION
**Pull request by kaise-lafrai**
## Description

See _issueid_. https://app.zenhub.com/workspaces/vagov-cms-team-5c0e7b864b5806bc2bfc2087/issues/department-of-veterans-affairs/va.gov-cms/2323

## QA steps

As user admin
1. Login

1. Then validate Acceptance Criteria from issue
- [ ] Confirm new paragraph type: "Step"
- [ ] paragraph "Step-by-Step" - single cardinality
- [ ] paragraph "Step" - multiple cardinality
- [ ] Confirm "Step" has the following fields: "field_media, field_wysiwyg"
  - [ ] field_wysiwyg is open by default and required - single cardinality
  - [ ] field_media is optional and presented as a button "Add image" - single cardinality
  - [ ] field_media has the following help text: "This image will be displayed at its full width, with no upscaling, without being cropped."

- [ ]  spec tool is updated w/ new paragraph spec, bundle, media image style and media image effect.

## Definition of Done
- [ ] Product release are written (or in progress), if required.
- [ ] Documentation has been updated, if applicable.

### CODE REVIEW

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 1]
```diff
@@ -3,67 +3,69 @@ Feature: Content model bundles
   In order to enter structured content into my site
   As a content editor
   I want to have content entity bundles that reflect my content model.
+  @dst @content_type @dstbundles
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:6]: Moderate: Something's not right with the formatting.  I should be seeing only the changes to the lines you changed.

File: tests/behat/drupal-spec-tool/content_model_paragraph_fields.feature [hunk 2]
```diff
@@ -4,61 +4,66 @@ Feature: Content model: Paragraph fields
   As a content editor
   I want to have paragraph fields that reflect my content model.
 
-  @dst @field_type @paragraph_fields @dstfields
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_paragraph_fields.feature:7]: Moderate: Shouldn't be seeing a change to every line.  Only the entries you altered.  You may have to tweak around with any behat related extensions you have in vscode to not mess with formatting of .feature files.

File: tests/behat/drupal-spec-tool/media.feature [hunk 3]
```diff
@@ -4,7 +4,7 @@ Feature: Media
   As a site owner
   I want to have image styles for various contexts.
 
-  @dst @image_styles
+  @dst @image_styles
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/media.feature:7]: Shouldn't be all the following spaces.

File: tests/behat/drupal-spec-tool/media.feature [hunk 4]
```diff
@@ -25,9 +25,10 @@ Feature: Media
 | Cropped: Square    | crop_square  |
 | Full content width with no upscaling | full_content_width |
 | Linkit result thumbnail  | linkit_result_thumbnail  |
+| Original | original |
 | 2:3 medium thumbnail  | 2_3_medium_thumbnail  |
 
-  @dst @image_effects
+  @dst @image_effects
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/media.feature:31]: Following spaces?

**Comment by kaise-lafrai** [comment]
@swirtSJW Good catch, not sure why I didn't notice that before. I will resolve.

File: config/sync/field.field.paragraph.step_by_step.field_add_step.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+status: true
+dependencies:
+  config:
+    - field.storage.paragraph.field_add_step
+    - paragraphs.paragraphs_type.step
+    - paragraphs.paragraphs_type.step_by_step
+  module:
+    - entity_reference_revisions
+id: paragraph.step_by_step.field_add_step
+field_name: field_add_step
```
Comment (oksana-c) [inline config/sync/field.field.paragraph.step_by_step.field_add_step.yml:12]: I'd like to suggest a field machine name change - `field_add_step` => `field_step`. Reason: nouns over verbs.

**Comment by kaise-lafrai** [comment]
@oksana-c @swirtSJW Ready for re review.

**Comment by kaise-lafrai** [comment]
> @kaise-lafrai , one last change - let's remove field_markup and place the text you added as markup value into image field help text. I understand the rationale for that field is to place help text above the field, but let's avoid non-functional fields in content model. We'll be addressing help text positioning separately as a part of larger effort. Thank you

Sounds good.

**Comment by kaise-lafrai** [comment]
@oksana-c Updated PR removed field markup and moved it to image help test. Also updated spec tool tests.

**Pull request merged**